components/nexus-repository/src/test/java/org/sonatype/nexus/repository/storage/StorageFacetImplIT.java

```diff
@@ -561,4 +604,160 @@ public void duplicateAssetName() throws Exception {
+      tx.saveAsset(asset);
+
+      final Iterable<Asset> assets = tx.browseAssets(component);
+      final Asset reloadedAsset = assets.iterator().next();
+
+      reloadedAsset.formatAttributes().set("attribute2", "alternate");
+      tx.saveAsset(reloadedAsset);
+
+      tx.commit();
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.findComponentWithProperty("version", "testVersion", tx.getBucket());
+      final Iterable<Asset> assets = tx.browseAssets(component);
+      final Asset asset = assets.iterator().next();
+
+      assertThat(asset.formatAttributes().get("attribute1", String.class), equalTo("original"));
+      assertThat(asset.formatAttributes().get("attribute1", String.class), equalTo("original"));
+    }
+  }
+
+  @Test
+  public void transactionsRollBackWhenRequired() throws Exception {
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.createComponent(tx.getBucket(), testFormat)
+          .group("myGroup")
+          .version("0.9")
+          .name("myComponent");
+      tx.saveComponent(component);
+      tx.rollback();
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.findComponentWithProperty("group", "myGroup", tx.getBucket());
+      assertThat(component, is(nullValue()));
+    }
+  }
+
+  @Test
+  public void transactionsRollBackOnException() throws Exception {
+    try {
+      try (StorageTx tx = underTest.openTx()) {
+        final Component component = tx.createComponent(tx.getBucket(), testFormat)
+            .group("myGroup")
+            .version("0.9")
+            .name("myComponent");
+        tx.saveComponent(component);
+        if (true) {
+          throw new MidTransactionException();
+        }
+      }
+    }
+    catch (MidTransactionException ignored) {
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.findComponentWithProperty("group", "myGroup", tx.getBucket());
+      assertThat(component, is(nullValue()));
+    }
+  }
+
+  @Test
+  public void transactionContentIsSaved() throws Exception {
+    try (StorageTx tx = underTest.openTx()) {
+      final Component component = tx.createComponent(tx.getBucket(), testFormat)
+          .group("myGroup")
+          .version("0.9")
+          .name("myComponent");
+      tx.saveComponent(component);
+      tx.commit();
+    }
+
+    try (StorageTx tx = underTest.openTx()) {
+      final Iterable<Component> components = tx.browseComponents(tx.getBucket());
+      for (Component c : components) {
+        System.err.println(c);
+      }
+
+      final Component component = tx.findComponentWithProperty("group", "myGroup", tx.getBucket());
+      assertThat(component, is(notNullValue()));
+      assertThat(component.group(), is("myGroup"));
+    }
+  }
+
+  @Test
+  public void entityIdCanBeUsedInLaterTransactions() throws Exception {
+    EntityId componentId = null;
```

Comment:
no need for explicitly setting null, and then IDEA will stop flagging them too :)